tests/e2e_tests/utils.py

```
@@ -0,0 +1,98 @@
+                print(f"Extracted {archive_file} to {extract_to}")
+
+        else:
+            raise ValueError(f"Unsupported archive format: {archive_file}")
+
+
+    except (OSError, ValueError, tarfile.TarError, zipfile.BadZipFile) as e:
+        print(f"Failed to extract archive {archive_file}: {e}")
+        # Try fallback to subprocess for tar files
+        if file_name.endswith(('.tar.xz', '.tar.gz', '.tar.bz2', '.tgz', '.tbz2', '.tar')):
+            try:
+                print(f"Trying fallback extraction with tar command...")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)